### ISSUE TITLE
Sensitive value in the result

### ISSUE THREAD

**Issue by mikluko** [loafoe/terraform-provider-ssh#45]
I have a use case where the resulting value is sensitive. But theres no way to mark it as such and upon destruction I see the following in the plan:

```
- resource "ssh_resource" "token" {
      - agent                       = false -> null
      - bastion_port                = "22" -> null
      - commands                    = [
          - "cat /some/path/to/sensitive/file/here",
        ] -> null
      - commands_after_file_changes = true -> null
      - host                        = "xx.xx.xx.xx" -> null
      - id                          = "5577006791947779410" -> null
      - port                        = "22" -> null
      - private_key                 = (sensitive value)
      - result                      = <<-EOT
<<<sensitive value in clear text here>>>
EOT -> null
      - timeout                     = "5m" -> null
      - user                        = "root" -> null
      - when                        = "create" -> null
    }
```

So this issue is bassically a feature proposal.

To my best knowledge, there is no way in Terraform to make value optionally sensitive. But I see two distinct takes on the resolution.

1. Follow (https://registry.terraform.io/providers/hashicorp/local/latest/docs) approach which implements separate `local_file` and `local_sensitive_file`. E.g. implement `ssh_sensitive_resource` which would behave like `ssh_resource` in all regards except for `result` property returning sensitive value.

2. Add `sensitive` (r/w, default: `false`) and `sensitive_result` (r/o, sensitive) properties to `ssh_resource`. If `sensitive` is `true`, then populate `sensitive_result` instead of `result`.

Later seems cleaner to me, although former follows existing pattern. 

@loafoe Let me know what you like better and if you would be willing to accept pull request.

**Comment by loafoe** [comment]
@mikluko looking at the local provider it seems they used option 2 and then decided to go for separate resources. There's probably a story behind that I'm leaning towards going with a separate sensitive resource. I'll give it a go shortly

**Comment by loafoe** [comment]
@mikluko implemented in `v2.2.0` of the provider
